src/app/modules/item/components/user-progress/user-progress.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { Progress } from '../../pages/item-chapter-view-group-situation/item-chapter-view-group-situation.component';
+
+@Component({
+  selector: 'alg-user-progress',
+  templateUrl: './user-progress.component.html',
+  styleUrls: [ './user-progress.component.scss' ]
+})
+export class UserProgressComponent implements OnChanges {
+
+  @Input() userProgress?: Progress;
+  @Input() canAccess?: boolean;
+
+  @Output() access = new EventEmitter<void>();
+
+  state: 'success'|'in-progress'|'no-score'|'not-started' = 'no-score';
+
+  ngOnChanges(_changes: SimpleChanges): void {
+    if (!this.userProgress) return;
+
+    if (this.userProgress.validated || this.userProgress.score === 100) this.state = 'success';
+    else if (this.userProgress.score > 0) this.state = 'in-progress';
+    else if (this.userProgress.score === 0 && this.userProgress.timeSpent > 0) this.state = 'no-score';
+    else this.state = 'not-started';
+  }
+
+  onAccess(): void {
+    this.access.emit();
```

Comment:
From the point of view of this component, it is "onAccess" (or "onAccessClick") makes sense as its button is called "access", and renaming the button may result in renaming the event with only impact to the component itself.
But for the parent components, emitting "access" is quite unclear. "permEditionRequested" for instance may be clearer for the component using this one, and so for the developers reading its code.

Reply:
fixed